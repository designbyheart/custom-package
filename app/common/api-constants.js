// @flow
// TODO:KS Consider moving it to api folder
export const TOKEN_EXPIRED_CODE = 'CRS-121'
export const SERVER_ERROR_CODE = 'OCS-000'
export const SERVER_API_CALL_ERROR = 'OCS-001'
export const PENDING_CONNECTION_REQUEST_CODE = 'CRS-103'
export const PUSH_NOTIFICATION_SENT_CODE = 'push-notification-sent'

export const QR_CODE_SIGNATURE = 's'
export const QR_CODE_CHALLENGE = 'c'
export const QR_CODE_REMOTE_CONNECTION_LOGO = 'lg'
export const QR_CODE_TARGET_DID = 'tDID'
export const QR_CODE_REMOTE_HOSTING_DID = 'rhDID'
export const QR_CODE_REMOTE_PAIR_WISE_DID = 'rpDID'
export const QR_CODE_TARGET_NAME = 'tn'
export const QR_CODE_SENDER_NAME = 'sn'
export const QR_CODE_UID = 'uid'

export const PUSH_NOTIFICATION_TYPE = {
  AUTH: 'auth-req',
  CLAIM_OFFER: 'claim-offer',
}
