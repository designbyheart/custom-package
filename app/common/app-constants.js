import { Platform } from 'react-native'

export const apptentiveCredentials = Platform.select({
  ios: {
    apptentiveKey: 'IOS-EVERNYM-ee7e2325084e',
    apptentiveSignature: '8049eba0f656a5b7aeb9722b84e5ec54',
  },
  android: {
    apptentiveKey: '<YOUR_ANDROID_APPTENTIVE_KEY>',
    apptentiveSignature: '<YOUR_ANDROID_APPTENTIVE_SIGNATURE>',
  },
})
