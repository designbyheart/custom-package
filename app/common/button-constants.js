export const ALLOW = 'Allow'
export const DENY = 'Deny'
