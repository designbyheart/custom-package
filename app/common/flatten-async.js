// @flow

// TODO:KS
// Explain the need for this function

export const flattenAsync = (
  fn: (...variableArgs: any[]) => Promise<any>
) => async (
  ...args: Array<any>
): Promise<[null | typeof Error, null | any]> => {
  try {
    const response = await fn(...args)
    return [null, response]
  } catch (e) {
    return [e, null]
  }
}
