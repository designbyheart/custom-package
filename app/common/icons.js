// @flow
import React from 'react'
import { Icon } from 'react-native-eva-icons'
import { colors } from './styles/constant'
import { moderateScale } from 'react-native-size-matters'

type IconProps = {
  width?: number,
  height?: number,
  color?: string,
  style?: any,
}

type CommonIconProps = {
  name: string,
} & IconProps

export const HOME_ICON = 'home-outline'
export const CONNECTIONS_ICON = 'people-outline'
export const CREDENTIALS_ICON = 'home-outline'
export const SETTINGS_ICON = 'settings-2-outline'

export const CAMERA_ICON = 'camera-outline'
export const CHECK_MARK_ICON = 'checkmark-circle-2-outline'
export const HOME_MENU_ICON = 'menu-outline'
export const ANDROID_BACK_ARROW_ICON = 'arrow-back-outline'
export const IOS_BACK_ARROW_ICON = 'arrow-ios-back-outline'
export const MORE_ICON = 'more-vertical-outline'
export const DELETE_ICON = 'trash-2-outline'
export const CLOSE_ICON = 'close-outline'

// common icon class is implemented to set default values (except name) for icons
// in that case we should only set icon name and get an icon with default color and size
export const EvaIcon = (props: CommonIconProps) => {
  const { name, width, height, color, style } = props

  return (
    <Icon
      name={name}
      width={width ? width : moderateScale(22)}
      height={height ? height : moderateScale(22)}
      fill={color ? color : colors.cmGray2}
      style={style}
    />
  )
}
