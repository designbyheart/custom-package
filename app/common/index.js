// @flow
export * from './api-constants'
export * from './button-constants'
export * from './message-constants'
export * from './route-constants'
export * from './secure-storage-constants'
export function noop() {}
