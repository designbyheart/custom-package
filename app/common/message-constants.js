// @flow
// for overall app message string, it can be error, success, warning etc.

export const TOUCH_ID_MESSAGE = 'Please confirm with TouchID'
export const TOUCH_ID_NOT_AVAILABLE = 'Touch ID is not supported on this device'

export const DEVICE_ENROLLMENT_ERROR = [
  'Enrollment Error',
  'Identifier or Seed or Push-notification token not present!',
]

export const PUSH_NOTIFICATION_PERMISSION_ERROR = [
  'Permission Error',
  'Push Notification permission needed! Please enable it from Settings.',
]

export const DUPLICATE_CONNECTION_ERROR = [
  'Attention!',
  'Duplicate Connection invitation!',
]

export const UNLOCKING_APP_WAIT_MESSAGE = 'Unlocking the App, please wait...'

export const ENTER_PASS_CODE_MESSAGE = 'Enter pass code'

export const ENTER_YOUR_PASS_CODE_MESSAGE = 'Enter your pass code'
