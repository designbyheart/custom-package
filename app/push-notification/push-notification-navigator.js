// @flow
import React, { PureComponent } from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import { PUSH_NOTIFICATION_TYPE } from '../services'
import {
  fetchAdditionalData,
  authenticationRequestReceived,
  claimOfferReceived,
  proofRequestReceived,
  pushNotificationReceived,
  addPendingRedirection,
} from '../store'
import {
  claimOfferRoute,
  invitationRoute,
  proofRequestRoute,
  qrCodeScannerTabRoute,
} from '../common'

import type {
  NotificationPayload,
  PushNotificationNavigatorProps,
  NextPropsPushNotificationNavigator,
} from './type-push-notification'

const blackListedRoute = {
  [invitationRoute]: invitationRoute,
  [qrCodeScannerTabRoute]: qrCodeScannerTabRoute,
}

export class PushNotificationNavigator extends PureComponent<
  void,
  PushNotificationNavigatorProps,
  void
> {
  componentWillReceiveProps(nextProps: PushNotificationNavigatorProps) {
    if (
      nextProps.pushNotification.notification &&
      nextProps.pushNotification.notification !==
        this.props.pushNotification.notification
    ) {
      const {
        type,
        additionalData,
        uid,
        senderLogoUrl,
        remotePairwiseDID,
      } = nextProps.pushNotification.notification

      if (type)
        if (!blackListedRoute[this.props.currentScreen])
          switch (type) {
            case PUSH_NOTIFICATION_TYPE.AUTH:
              // TODO:PS: handle auth request
              break

            case PUSH_NOTIFICATION_TYPE.CLAIM_OFFER:
              this.props.claimOfferReceived(additionalData, {
                uid,
                senderLogoUrl,
                remotePairwiseDID,
              })
              this.handleRedirection(claimOfferRoute, {
                uid,
              })
              break

            case PUSH_NOTIFICATION_TYPE.PROOF_REQUEST:
              this.props.proofRequestReceived(additionalData, {
                uid,
                senderLogoUrl,
                remotePairwiseDID,
              })
              this.handleRedirection(proofRequestRoute, {
                uid,
              })
              break
          }
    }
  }

  //TODO:PS: write type for params
  handleRedirection = (routeName: string, params: any) => {
    if (this.props.isAppLocked)
      this.props.addPendingRedirection(routeName, params)
    else this.props.navigateToRoute(routeName, params)
  }

  render() {
    return null
  }
}

const mapStateToProps = ({ pushNotification, route, lock }) => ({
  pushNotification,
  currentScreen: route.currentScreen,
  isAppLocked: lock.isAppLocked,
})

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      fetchAdditionalData,
      authenticationRequestReceived,
      claimOfferReceived,
      proofRequestReceived,
      pushNotificationReceived,
      addPendingRedirection,
    },
    dispatch
  )

export default connect(mapStateToProps, mapDispatchToProps)(
  PushNotificationNavigator
)
