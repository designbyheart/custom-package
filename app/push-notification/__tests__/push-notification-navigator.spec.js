// @flow
import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { homeTabRoute } from '../../common'
import { PushNotificationNavigator } from '../push-notification-navigator'

describe('<PushNotificationNavigator />', () => {
  let store = {}

  const storeToProps = {
    pushNotification: {
      notification: {
        additionalData: {
          data: {
            name: 'Home Address',
            version: '1.0.0',
            revealedAttributes: [
              {
                label: 'Address 1',
                data: 'Address Address Address',
              },
              {
                label: 'Address 2',
                data: 'Address 2 Address 2 Address 2',
              },
            ],
          },
          issuer: {
            name: 'Test Issuer',
          },
          statusMsg: 'pending',
        },
        type: 'claimOffer',
        uid: 'usd123',
        senderLogoUrl: 'http://testissuer.com/logoUrl.png',
        remotePairwiseDID: 'ha66899sadfjZJGINKN0770',
      },
    },
    currentScreen: homeTabRoute,
    isAppLocked: false,
  }

  beforeAll(() => {
    store = {
      getState() {
        return {
          ...storeToProps,
        }
      },
      subscribe() {
        return jest.fn()
      },
      dispatch() {
        return jest.fn()
      },
    }
  })

  function props() {
    return {
      ...storeToProps,
      navigateToRoute: jest.fn(),
      fetchAdditionalData: jest.fn(),
      authenticationRequestReceived: jest.fn(),
      claimOfferReceived: jest.fn(),
      proofRequestReceived: jest.fn(),
      pushNotificationReceived: jest.fn(),
      addPendingRedirection: jest.fn(),
    }
  }

  it('should render PushNotification components properly', () => {
    const wrapper = renderer
      .create(
        <Provider store={store}>
          <PushNotificationNavigator {...props()} />
        </Provider>
      )
      .toJSON()
    expect(wrapper).toMatchSnapshot()
  })
})
