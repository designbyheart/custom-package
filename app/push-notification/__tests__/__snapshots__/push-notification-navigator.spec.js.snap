// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PushNotificationNavigator /> should render PushNotification components properly 1`] = `null`;
